### DISCUSSION TITLE
The future of open-source delayed message exchange plugin

### DISCUSSION THREAD

**Discussion by gomoripeti** [rabbitmq/rabbitmq-delayed-message-exchange#287] [General]
The RabbitMQ Core Team has no interest in maintaining this plugin (and rather spend effort on a commercial alternative solution https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/discussions/286#discussioncomment-10629880, this is totally acceptable) and the plugin will need at least a short-term solution to work with future RabbitMQ versions that use Khepri metadata store (https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/issues/272) and a longer-term solution to address the many limitations (https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/issues/229)

How would the Core Team like to handle this repository and the open-source plugin
- would the repo stay under the rabbitmq github org? would the Core Team still accept PRs and release new versions?
- or should the maintainers be handed over (maybe to a community team?)?
- or should this repo be marked as discontinued?

We at CloudAMQP are willing to continue contributing or taking part in maintaining the plugin. A short-term solution would definitely be useful for us and I am sure that the Core Team's commercial implementation will be superior to a long-term solution in the open-source version so it won't be competition.

What is AmazonMQ team's (@SimonUnge) stance on this plugin (after the initial draft for a long-term solution https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/pull/253)?

I appreciate any information, opinions and how to go forward.

**Comment by michaelklishin** [comment]
As I said in #272, Mnesia is not removed in 4.0 and should still be available for this plugin. So there should be little to no short term impact.

If someone wants to maintain it in the 4.x era, the core team won't object to that. We understand that with all the shortcomings, it is a popular plugin that should be functional with its current design.
